chains/solana/contracts/programs/ccip-offramp/src/instructions/v1/commit.rs

```diff
@@ -230,6 +231,74 @@ impl Commit for Impl {
 
         Ok(())
     }
+
+    fn close_commit_report_account(
+        &self,
+        ctx: Context<CloseCommitReportAccount>,
+        _source_chain_selector: u64,
+        _root: Vec<u8>,
+    ) -> Result<()> {
+        let commit_report = &ctx.accounts.commit_report;
+
+        // Check if all messages have been executed (must be Success)
+        require!(
+            all_messages_executed(commit_report),
+            CcipOfframpError::CommitReportHasPendingMessages
+        );
+
+        // Close the account and convert rent to wrapped SOL
+        transfer_and_wrap_native_sol(
+            &ctx.accounts.token_program,
+            ctx.accounts.commit_report.to_account_info(),
+            &ctx.accounts.fee_token_receiver,
+        )?;
+
+        let merkle_root_array: [u8; 32] = _root
+            .try_into()
+            .map_err(|_| CcipOfframpError::InvalidProof)?;
+
+        emit!(CommitReportPDAClosed {
+            source_chain_selector: _source_chain_selector,
+            merkle_root: merkle_root_array,
+        });
+
+        Ok(())
+    }
+}
+
+// Helper function to check if all messages have been executed
+fn all_messages_executed(report: &CommitReport) -> bool {
+    let num_messages = report.max_msg_nr.saturating_sub(report.min_msg_nr) + 1;
+
+    // execution_states follow geometric series 2^1 + 2^3 + 2^5 + ... + 2 * 2^(2 * (num_messages - 1))
+    // it can be converted to 2 * (4^0 + 4^1 + 4^2 + ... + 4^(num_messages - 1))
+    // sum is calculated as 2 * (4^num_messages - 1) / 3
+    let fully_executed = (4u128.pow(num_messages as u32) - 1) * 2 / 3;
+    report.execution_states == fully_executed
+}
+
+// Helper function to convert the SOL from the closed account to wrapped SOL
+fn transfer_and_wrap_native_sol<'info>(
+    token_program: &Program<'info, Token>,
+    commit_report: AccountInfo<'info>,
+    fee_token_receiver: &Account<'info, TokenAccount>,
+) -> Result<()> {
+    // Get lamports before closing
+    let lamports = commit_report.lamports();
+
+    // Close the account by setting its data length to 0
```

Comment:
This comment seems outdated, you're closing it by just removing its rent-exempt balance